### PR TITLE
x11-wm/cwm updated to latest 6.6 and new proxy-maintainer

### PR DESCRIPTION
**Pull request by epsilon-0**
Package has been stale and using out of date version from 7 years ago.

Adding myself as proxy-maintainer

Bug: https://bugs.gentoo.org/718682

Signed-off-by: Aisha Tammy <gentoo@aisha.cc>

### CODE REVIEW

File: x11-wm/cwm/cwm-6.6.ebuild [hunk 1]
```diff
@@ -1,35 +1,34 @@
...
-SRC_URI="vanilla? ( http://chneukirchen.org/releases/${P}.tar.gz -> ${P}-chneukirchen.tar.gz )
-	!vanilla? ( https://github.com/xmw/cwm/tarball/ea9a436 -> ${P}-xmw.tar.gz )"
+
+if [[ ${PV} == 9999 ]]; then
+	inherit git-r3
+	EGIT_REPO_URI="https://github.com/chneukirchen/${PN}.git"
+	KEYWORDS=""
+else
+	SRC_URI="https://github.com/chneukirchen/${PN}/archive/v${PV}.tar.gz -> ${P}.tar.gz"
+	KEYWORDS="amd64 arm64 x86"
```
Comment (gyakovlev) [inline x11-wm/cwm/cwm-6.6.ebuild:18]: you can't add new ebuild with stable keywords

File: x11-wm/cwm/cwm-6.6.ebuild [hunk 2]
```diff
@@ -1,35 +1,34 @@
...
-
-src_unpack() {
-	default
-	# vcs-snapshot doesn't work with tarball names
-	if ! use vanilla ; then
-		mv *${PN}-* ${P} || die
-	fi
-}
+	sys-devel/bison
+"
```
Comment (gyakovlev) [inline x11-wm/cwm/cwm-6.6.ebuild:31]: those dependencies are usually BDEPEND in EAPI=7

File: x11-wm/cwm/cwm-6.6.ebuild [hunk 3]
```diff
@@ -1,35 +1,34 @@
-# Copyright 1999-2018 Gentoo Authors
+# Copyright 2020 Gentoo Authors
```
Comment (gyakovlev) [inline x11-wm/cwm/cwm-6.6.ebuild:1]: you should leave original year intact.

File: x11-wm/cwm/cwm-6.6.ebuild [hunk 4]
```diff
@@ -1,35 +1,34 @@
...
 # Distributed under the terms of the GNU General Public License v2
 
-EAPI=5
+EAPI=7
 
-inherit eutils toolchain-funcs
+inherit desktop eutils toolchain-funcs
```
Comment (gyakovlev) [inline x11-wm/cwm/cwm-6.6.ebuild:6]: I don't see eutils being used in the ebuild

File: x11-wm/cwm/cwm-6.6.ebuild [hunk 5]
```diff
@@ -1,35 +1,34 @@
...
 
 DESCRIPTION="OpenBSD fork of calmwm, a clean and lightweight window manager"
 HOMEPAGE="https://www.openbsd.org/cgi-bin/cvsweb/xenocara/app/cwm/
 	https://github.com/chneukirchen/cwm"
-SRC_URI="vanilla? ( http://chneukirchen.org/releases/${P}.tar.gz -> ${P}-chneukirchen.tar.gz )
-	!vanilla? ( https://github.com/xmw/cwm/tarball/ea9a436 -> ${P}-xmw.tar.gz )"
+
+if [[ ${PV} == 9999 ]]; then
+	inherit git-r3
+	EGIT_REPO_URI="https://github.com/chneukirchen/${PN}.git"
```
Comment (gyakovlev) [inline x11-wm/cwm/cwm-6.6.ebuild:14]: this url is incorrect (as is down below)
It's https://github.com/leahneukirchen/cwm

**Comment by jonasstein** [comment]
Please change
Bug: https://bugs.gentoo.org/718682
to
Closes: https://bugs.gentoo.org/718682

**Comment by epsilon-0** [comment]
> We don't usually remove old ebuilds on big bumps. Leave old ebuild in place and introduce new one and live as new files.
> 

I've added the old one back :)

> Also makefile calls pkg-config directly, it'll break cross-compilation.
> Will be nicer if calls to pkg-config will be replaced by proper call.
> for example, modify Makefile to add
> 
> `PKG_CONFIG?= pkg-coinfig`
> 
> and have ebuild define it as `PKG_CONFIG="$(tc-getPKG_CONFIG)"` , similar to how `CC` is defined on command line and in Makefile.

I've made a pull request upstream so that this will be easier next release onwards.
Until then, I've added a patch, hopefully that should be OK :)

Thanks a lot for all your comments @gyakovlev

**Comment by epsilon-0** [comment]
> Please change
> Bug: https://bugs.gentoo.org/718682
> to
> Closes: https://bugs.gentoo.org/718682

I've changed that to the correct one, thanks a lot for the information @jonasstein :)

File: x11-wm/cwm/cwm-6.6.ebuild (hunk 5 above)
Comment (epsilon-0) [reply x11-wm/cwm/cwm-6.6.ebuild]: I've updated the link, thanks a lot :)

File: x11-wm/cwm/cwm-6.6.ebuild (hunk 4 above)
Comment (epsilon-0) [reply x11-wm/cwm/cwm-6.6.ebuild]: I've removed it from the ebuild, thanks again

File: x11-wm/cwm/cwm-6.6.ebuild (hunk 3 above)
Comment (epsilon-0) [reply x11-wm/cwm/cwm-6.6.ebuild]: I've updated the years to the correct range, hopefully it is OK now.

File: x11-wm/cwm/cwm-6.6.ebuild (hunk 2 above)
Comment (epsilon-0) [reply x11-wm/cwm/cwm-6.6.ebuild:31]: Totally forgot, thanks for noticing, I've fixed that :)

File: x11-wm/cwm/cwm-6.6.ebuild (hunk 1 above)
Comment (epsilon-0) [reply x11-wm/cwm/cwm-6.6.ebuild]: Yep, I've changed it back to the unstable ones, and I've also readded the `~arm` keyword I removed by mistake.

File: x11-wm/cwm/cwm-6.6.ebuild [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+LICENSE="ISC"
+SLOT="0"
+
+RDEPEND="x11-libs/libXft
+	x11-libs/libXinerama
+	x11-libs/libXrandr
+"
+BDEPEND="${RDEPEND}
+	virtual/pkgconfig
+	sys-devel/bison
```
Comment (gyakovlev) [inline x11-wm/cwm/cwm-6.6.ebuild:30]: small nit, sort alphabetically. I can do it on merge.

File: x11-wm/cwm/cwm-6.6.ebuild [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+fi
+
+LICENSE="ISC"
+SLOT="0"
+
+RDEPEND="x11-libs/libXft
+	x11-libs/libXinerama
+	x11-libs/libXrandr
+"
+BDEPEND="${RDEPEND}
```
Comment (gyakovlev) [inline x11-wm/cwm/cwm-6.6.ebuild:28]: no, this is incorrect.
you should define
DEPEND="${RDEPEND}",
BDEPEND should only contain bison and pkgconfig

Comment (gyakovlev) [reply x11-wm/cwm/cwm-6.6.ebuild:28]: here's some documentation on differences between DEPEND and BDEPEND
https://dev.gentoo.org/~mgorny/articles/the-ultimate-guide-to-eapi-7.html#build-time-dependencies-split-into-depend-and-bdepend

Comment (epsilon-0) [reply x11-wm/cwm/cwm-6.6.ebuild]: Thanks a lot for the link!

I've split them correctly.

File: x11-wm/cwm/cwm-6.6.ebuild (hunk 6 above)
Comment (epsilon-0) [reply x11-wm/cwm/cwm-6.6.ebuild]: Haha, I've sorted them. Should look cleaner now :)

**Comment by gyakovlev** [comment]
I merged it with some minor fixes, closing manually. thanks!

**Comment by gyakovlev** [comment]
https://github.com/gentoo/gentoo/commit/e84ae9943a70dcac2434c18533327df73b471a71

**Pull request closed without merging**